diff -Nur -x '*.orig' -x '*~' ldapscripts/Makefile ldapscripts.new/Makefile
--- ldapscripts/Makefile	2009-06-22 20:00:40.420731145 +0400
+++ ldapscripts.new/Makefile	2009-06-22 20:01:58.784731571 +0400
@@ -31,6 +31,7 @@
 MANDIR = $(PREFIX)/man
 ETCDIR = $(PREFIX)/etc/$(NAME)
 LIBDIR = $(PREFIX)/lib/$(NAME)
+DESTDIR = ""
 
 # Files to install
 RUNFILE = runtime
@@ -91,91 +92,91 @@
 # Install targets
 install:	installsbin installman installetc installlib
 installsbin:	configure
-	@echo -n 'Installing scripts into $(SBINDIR)... '
-	@install -d -m 755 '$(SBINDIR)' 2>/dev/null
+	@echo -n 'Installing scripts into $(DESTDIR)$(SBINDIR)... '
+	@install -d -m 755 '$(DESTDIR)$(SBINDIR)' 2>/dev/null
 	@for i in $(SBINFILES) ; do \
-		install -m 750 "sbin/$$i.patched" "$(SBINDIR)/$$i" ; \
+		install -m 750 "sbin/$$i.patched" "$(DESTDIR)$(SBINDIR)/$$i" ; \
 	done
 	@echo 'ok.'
 
 installman:
-	@echo -n 'Installing man files into $(MANDIR)... '
-	@install -d -m 755 '$(MANDIR)/man1' 2>/dev/null
+	@echo -n 'Installing man files into $(DESTDIR)$(MANDIR)... '
+	@install -d -m 755 '$(DESTDIR)$(MANDIR)/man1' 2>/dev/null
 	@for i in $(MAN1FILES) ; do \
-		cat "man/man1/$$i" | gzip - > "$(MANDIR)/man1/`basename $$i`.gz" ; \
+		cat "man/man1/$$i" | gzip - > "$(DESTDIR)$(MANDIR)/man1/`basename $$i`.gz" ; \
 	done
-	@install -d -m 755 '$(MANDIR)/man5' 2>/dev/null
+	@install -d -m 755 '$(DESTDIR)$(MANDIR)/man5' 2>/dev/null
 	@for i in $(MAN5FILES) ; do \
-		cat "man/man5/$$i" | gzip - > "$(MANDIR)/man5/`basename $$i`.gz" ; \
+		cat "man/man5/$$i" | gzip - > "$(DESTDIR)$(MANDIR)/man5/`basename $$i`.gz" ; \
 	done
 	@echo 'ok.'
 
 installetc:	configure
-	@echo -n 'Installing configuration files into $(ETCDIR)... '
-	@install -d -m 755 '$(ETCDIR)' 2>/dev/null
-	@install -m 440 'etc/$(ETCFILE).patched' '$(ETCDIR)/$(ETCFILE).sample'
-	@if [ ! -f '$(ETCDIR)/$(ETCFILE)' ]; then \
-		install -m 640 '$(ETCDIR)/$(ETCFILE).sample' '$(ETCDIR)/$(ETCFILE)'; \
+	@echo -n 'Installing configuration files into $(DESTDIR)$(ETCDIR)... '
+	@install -d -m 755 '$(DESTDIR)$(ETCDIR)' 2>/dev/null
+	@install -m 440 'etc/$(ETCFILE).patched' '$(DESTDIR)$(ETCDIR)/$(ETCFILE).sample'
+	@if [ ! -f '$(DESTDIR)$(ETCDIR)/$(ETCFILE)' ]; then \
+		install -m 640 '$(DESTDIR)$(ETCDIR)/$(ETCFILE).sample' '$(DESTDIR)$(ETCDIR)/$(ETCFILE)'; \
 	fi
-	@install -m 440 -b 'etc/$(PWDFILE)' '$(ETCDIR)/$(PWDFILE).sample'
-	@if [ ! -f '$(ETCDIR)/$(PWDFILE)' ]; then \
-		install -m 640 '$(ETCDIR)/$(PWDFILE).sample' '$(ETCDIR)/$(PWDFILE)'; \
+	@install -m 440 -b 'etc/$(PWDFILE)' '$(DESTDIR)$(ETCDIR)/$(PWDFILE).sample'
+	@if [ ! -f '$(DESTDIR)$(ETCDIR)/$(PWDFILE)' ]; then \
+		install -m 640 '$(DESTDIR)$(ETCDIR)/$(PWDFILE).sample' '$(DESTDIR)$(ETCDIR)/$(PWDFILE)'; \
 	fi
 	@for i in $(TMPLFILES) ; do \
-		install -m 440 "etc/$$i" '$(ETCDIR)' ; \
+		install -m 440 "etc/$$i" '$(DESTDIR)$(ETCDIR)' ; \
 	done
 	@echo 'ok.'
 
 installlib:	configure
-	@echo -n 'Installing library files into $(LIBDIR)... '
-	@install -d -m 755 '$(LIBDIR)' 2>/dev/null
-	@install -m 440 'lib/$(RUNFILE).patched' '$(LIBDIR)/$(RUNFILE)'
+	@echo -n 'Installing library files into $(DESTDIR)$(LIBDIR)... '
+	@install -d -m 755 '$(DESTDIR)$(LIBDIR)' 2>/dev/null
+	@install -m 440 'lib/$(RUNFILE).patched' '$(DESTDIR)$(LIBDIR)/$(RUNFILE)'
 	@echo 'ok.'
 
 # Uninstall targets
 deinstall: uninstall
 uninstall:	uninstallsbin uninstallman uninstalletc uninstalllib
 uninstallsbin:
-	@echo -n 'Uninstalling scripts from $(SBINDIR)... '
+	@echo -n 'Uninstalling scripts from $(DESTDIR)$(SBINDIR)... '
 	@for i in $(SBINFILES) ; do \
-		rm -f "$(SBINDIR)/$$i" ; \
+		rm -f "$(DESTDIR)$(SBINDIR)/$$i" ; \
 	done
-	@rmdir '$(SBINDIR)' 2>/dev/null || true
+	@rmdir '$(DESTDIR)$(SBINDIR)' 2>/dev/null || true
 	@echo 'ok.'
 
 uninstallman:
-	@echo -n 'Uninstalling man files from $(MANDIR)... '
+	@echo -n 'Uninstalling man files from $(DESTDIR)$(MANDIR)... '
 	@for i in $(MAN1FILES) ; do \
-		rm -f "$(MANDIR)/man1/`basename $$i`.gz" ; \
+		rm -f "$(DESTDIR)$(MANDIR)/man1/`basename $$i`.gz" ; \
 	done
-	@rmdir '$(MANDIR)/man1' 2>/dev/null || true
+	@rmdir '$(DESTDIR)$(MANDIR)/man1' 2>/dev/null || true
 	@for i in $(MAN5FILES) ; do \
-		rm -f "$(MANDIR)/man5/`basename $$i`.gz" ; \
+		rm -f "$(DESTDIR)$(MANDIR)/man5/`basename $$i`.gz" ; \
 	done
-	@rmdir '$(MANDIR)/man5' 2>/dev/null || true
+	@rmdir '$(DESTDIR)$(MANDIR)/man5' 2>/dev/null || true
 	@rmdir '$(MANDIR)' 2>/dev/null || true
 	@echo 'ok.'
 
 uninstalletc:
-	@echo -n 'Uninstalling configuration files from $(ETCDIR)... '
-	@if cmp -s '$(ETCDIR)/$(ETCFILE)' '$(ETCDIR)/$(ETCFILE).sample'; then \
-		rm -f '$(ETCDIR)/$(ETCFILE)'; \
+	@echo -n 'Uninstalling configuration files from $(DESTDIR)$(ETCDIR)... '
+	@if cmp -s '$(DESTDIR)$(ETCDIR)/$(ETCFILE)' '$(DESTDIR)$(ETCDIR)/$(ETCFILE).sample'; then \
+		rm -f '$(DESTDIR)$(ETCDIR)/$(ETCFILE)'; \
 	fi
-	@rm -f '$(ETCDIR)/$(ETCFILE).sample'
-	@if cmp -s '$(ETCDIR)/$(PWDFILE)' '$(ETCDIR)/$(PWDFILE).sample'; then \
-		rm -f '$(ETCDIR)/$(PWDFILE)'; \
+	@rm -f '$(DESTDIR)$(ETCDIR)/$(ETCFILE).sample'
+	@if cmp -s '$(DESTDIR)$(ETCDIR)/$(PWDFILE)' '$(DESTDIR)$(ETCDIR)/$(PWDFILE).sample'; then \
+		rm -f '$(DESTDIR)$(ETCDIR)/$(PWDFILE)'; \
 	fi
-	@rm -f '$(ETCDIR)/$(PWDFILE).sample'
+	@rm -f '$(DESTDIR)$(ETCDIR)/$(PWDFILE).sample'
 	@for i in $(TMPLFILES) ; do \
-		rm -f "$(ETCDIR)/$$i" ; \
+		rm -f "$(DESTDIR)$(ETCDIR)/$$i" ; \
 	done
-	@rmdir '$(ETCDIR)' 2>/dev/null || true
+	@rmdir '$(DESTDIR)$(ETCDIR)' 2>/dev/null || true
 	@echo 'ok.'
 
 uninstalllib:
-	@echo -n 'Uninstalling library files from $(LIBDIR)... '
-	@rm -f '$(LIBDIR)/$(RUNFILE)'
-	@rmdir '$(LIBDIR)' 2>/dev/null || true
+	@echo -n 'Uninstalling library files from $(DESTDIR)$(LIBDIR)... '
+	@rm -f '$(DESTDIR)$(LIBDIR)/$(RUNFILE)'
+	@rmdir '$(DESTDIR)$(LIBDIR)' 2>/dev/null || true
 	@echo 'ok.'
 
 # Clean targets
