diff -Nur -x '*.orig' -x '*~' ldapscripts/lib/runtime ldapscripts.new/lib/runtime
--- ldapscripts/lib/runtime	2009-06-24 15:55:24.968230982 +0400
+++ ldapscripts.new/lib/runtime	2009-06-24 15:55:54.500731327 +0400
@@ -624,7 +624,7 @@
 [ ! -e /dev/random ] && end_die  "You need a /dev/random special file to run these scripts"
 
 # Set USER variable to the real user (do not trust current $USER value)
-USER=`logname`
+USER=`id -nu`
 
 # Check for bindpwd file
 if [ ! -f "$BINDPWDFILE" ] || [ ! -r "$BINDPWDFILE" ]
