diff -Nur -x '*.orig' -x '*~' ldapscripts/etc/ldapscripts.conf ldapscripts.new/etc/ldapscripts.conf
--- ldapscripts/etc/ldapscripts.conf	2009-06-24 17:50:52.716231081 +0400
+++ ldapscripts.new/etc/ldapscripts.conf	2009-06-24 17:50:53.372231305 +0400
@@ -59,7 +59,8 @@
 # Command-line used to generate a password for added users (you may use %u for username here)
 # WARNING !!!! This is evaluated, everything specified here will be run !
 # Special value "<ask>" will ask for a password interactively
-PASSWORDGEN="head -c8 /dev/random | uuencode -m - | sed -n '2s|=*$||;2p' | sed -e 's|+||g' -e 's|/||g'"
+PASSWORDGEN="cat /dev/urandom|tr -dc 'a-zA-Z0-9'|head -c8"
+#PASSWORDGEN="head -c8 /dev/random | uuencode -m - | sed -n '2s|=*$||;2p' | sed -e 's|+||g' -e 's|/||g'"
 #PASSWORDGEN="echo changeme"
 #PASSWORDGEN="echo %u"
 #PASSWORDGEN="<ask>"
