Description: Fix the lastuser and lastmachine UID LDAP searches
Author: Slavko <linux@slavino.sk>
Index: ldapscripts/lib/runtime
===================================================================
--- ldapscripts.orig/lib/runtime	2011-03-21 17:23:05.000000000 +0300
+++ ldapscripts/lib/runtime	2011-03-21 17:23:30.000000000 +0300
@@ -529,7 +529,7 @@
 # Input : nothing
 # Output : the last machine id used + 1 (so the first useable machine id) ($_UID)
 _findlastmachine () {
-  _UID=`_ldapsearch "$SUFFIX" '(objectClass=posixAccount)' uidNumber | grep "uidNumber: " | sed "s|uidNumber: ||" | uniq | sort -n | tail -n 1`
+  _UID=`_ldapsearch "$MSUFFIX,$SUFFIX" '(objectClass=posixAccount)' uidNumber | grep "uidNumber: " | sed "s|uidNumber: ||" | uniq | sort -n | tail -n 1`
   if [ -z "$_UID" ] || [ ! "$_UID" -gt "$MIDSTART" ]
   then
     _UID="$MIDSTART"
@@ -541,7 +541,7 @@
 # Input : nothing
 # Output : the last user id used + 1 (so the first useable user id) ($_UID)
 _findlastuser () {
-  _UID=`_ldapsearch "$SUFFIX" '(objectClass=posixAccount)' uidNumber | grep "uidNumber: " | sed "s|uidNumber: ||" | uniq | sort -n | tail -n 1`
+  _UID=`_ldapsearch "$USUFFIX,$SUFFIX" '(objectClass=posixAccount)' uidNumber | grep "uidNumber: " | sed "s|uidNumber: ||" | uniq | sort -n | tail -n 1`
   if [ -z "$_UID" ] || [ ! "$_UID" -gt "$UIDSTART" ]
   then
     _UID="$UIDSTART"
